src/main/java/com/google/jepsenonspanner/verifier/knossos/Node.java

```diff
@@ -0,0 +1,272 @@
+package com.google.jepsenonspanner.verifier.knossos;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.jepsenonspanner.client.Record;
+import com.google.jepsenonspanner.operation.OpRepresentation;
+import us.bpsm.edn.Keyword;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.jepsenonspanner.client.Record.INVOKE_STR;
+import static com.google.jepsenonspanner.client.Record.OK_STR;
+
+/**
+ * This class encapsulates a node in the Linearizability DFS graph. Contains the state of the
+ * database, the records that are in progress and the ones that have been linearized. It
+ * is essentially the state of the current search. Corresponds to a config in the paper.
+ */
+public class Node {
+  private Map<String, Long> databaseState;
+  // Since each thread can have at most one record in both call and ret set, index them using pID
+  private HashMap<Long, Record> calls;
+  private HashMap<Long, Record> rets;
+  // Keeps track of the next position in the history to transition to
+  private int recordIdx;
+
+  // Memorize configs visited so far so that we do not go through an invalid path twice
+  private static HashSet<Node> configsVisited = new HashSet<>();
+  // Records the maximum record index we have seen; if we reached the end of the history, this
+  // should be equal to length of the history
+  private static int maxRecordIdxSeen = -1;
+
+  private Node(Map<String, Long> databaseState,
+               HashMap<Long, Record> calls, HashMap<Long, Record> rets, int recordIdx) {
+    this.databaseState = new HashMap<>(databaseState);
+    this.calls = new HashMap<>(calls);
+    this.rets = new HashMap<>(rets);
+    this.recordIdx = recordIdx;
+    if (recordIdx > maxRecordIdxSeen) {
+      maxRecordIdxSeen = recordIdx;
+    }
+  }
+
+  /**
+   * Constructor for the initial config.
```

Comment:
Since you changed the naming, you may want to search for every "config" and rename them.